src/game.js

```diff
@@ -209,6 +218,14 @@
             // reset the frame counter
             frameCounter = 0;
             frameRate = ~~(0.5 + 60 / me.sys.fps);
+            
+            // set step size based on the updatesPerSecond
+            stepSize = (1000 / Math.min(me.sys.updatesPerSecond, me.sys.fps));
```

Comment:
Let's try it.

* `me.sys.updatesPerSecond = 20;`
* `me.sys.fps = 60;`
* `stepSize = (1000 / 20);` stepSize will be 50.
* in `update`:
  * `accumulator += me.timer.getDelta();` accumulator =~ 16.6667
  * `updateDelta = stepSize` ... 50
  * `while (accumulator >= stepSize)` ... nope, not yet! Updates will happen later (body executes zero times)
  * later ... (exactly 3 frames)
  * `accumulator += me.timer.getDelta();` accumulator =~ 50.0001
  * `while (accumulator >= stepSize)` true!
    * `accumulator -= stepSize;` accumulator is now ~0.0001. The while loop runs only once.

Now let's rewind and run with `updatesPerSecond > fps`, just for giggles...

* `me.sys.updatesPerSecond = 200;`
* `me.sys.fps = 60;`
* `stepSize = (1000 / 60);` stepSize will be ~16.6667 (this is where capping to `fps` happens)
* in `update`:
  * `accumulator += me.timer.getDelta();` accumulator =~ 16.6667
  * `updateDelta = stepSize` ... ~16.6667
  * `while (accumulator >= stepSize)` true!
    * `accumulator -= stepSize;` accumulator is now ~0.0001. The while loop runs only once.

Hmmm...